sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -357,6 +357,28 @@ private static void init(final @NotNull SentryOptions options, final boolean glo
     }
   }
 
+  private static void initForOpenTelemetryMaybe(SentryOptions options) {
+    if (SentryOpenTelemetryMode.OFF == options.getOpenTelemetryMode()) {
+      options.setSpanFactory(new DefaultSpanFactory());
+      //    } else {
+      // enabling this causes issues with agentless where OTel spans seem to be randomly ended
```

Comment:
We have to keep using `SpanFactoryFactory` in the `SentryOptions` ctor for now as setting the `OtelSpanFactory` here causes OpenTelemetry spans to be ended in the agentless sample.

I haven't investigated yet why that's happening.

Reply:
Created https://github.com/getsentry/sentry-java/issues/4005 to track